src/test/java/com/amazonaws/encryptionsdk/keyrings/AwsKmsSymmetricKeyringTest.java

```diff
@@ -0,0 +1,279 @@
+        FAILING_KEY_ARN.getBytes(PROVIDER_ENCODING), generate(ALGORITHM_SUITE.getDataKeyLength()));
+    private static final KeyringTraceEntry ENCRYPTED_KEY_TRACE =
+        new KeyringTraceEntry(AWS_KMS_PROVIDER_ID, KEY_ARN, ENCRYPTED_DATA_KEY, SIGNED_ENCRYPTION_CONTEXT);
+    private static final KeyringTraceEntry GENERATED_DATA_KEY_TRACE =
+        new KeyringTraceEntry(AWS_KMS_PROVIDER_ID, KEY_ARN, GENERATED_DATA_KEY);
+    @Mock(lenient = true)
+    private DataKeyEncryptionDao dataKeyEncryptionDao;
+    private Keyring keyring;
+
+    @BeforeEach
+    void setup() {
+        when(dataKeyEncryptionDao.encryptDataKey(KEY_NAME, PLAINTEXT_DATA_KEY, ENCRYPTION_CONTEXT)).thenReturn(ENCRYPTED_KEY);
+        when(dataKeyEncryptionDao.decryptDataKey(ENCRYPTED_KEY, ALGORITHM_SUITE, ENCRYPTION_CONTEXT))
+            .thenReturn(new DecryptDataKeyResult(KEY_ARN, PLAINTEXT_DATA_KEY));
+
+        keyring = new AwsKmsSymmetricKeyring(dataKeyEncryptionDao, KEY_NAME);
+    }
+
+    @Test
+    void testMalformedArns() {
+        DecryptionMaterials decryptionMaterials = DecryptionMaterials.newBuilder()
+            .setAlgorithm(ALGORITHM_SUITE)
+            .setEncryptionContext(ENCRYPTION_CONTEXT)
+            .build();
+
+        List<EncryptedDataKey> encryptedDataKeys = new ArrayList<>();
+        encryptedDataKeys.add(new KeyBlob(AWS_KMS_PROVIDER_ID, "arn:badArn".getBytes(PROVIDER_ENCODING), new byte[]{}));
+        encryptedDataKeys.add(ENCRYPTED_KEY);
+
+        decryptionMaterials = keyring.onDecrypt(decryptionMaterials, encryptedDataKeys);
+        assertEquals(PLAINTEXT_DATA_KEY, decryptionMaterials.getCleartextDataKey());
+
+        decryptionMaterials = DecryptionMaterials.newBuilder()
+            .setAlgorithm(ALGORITHM_SUITE)
+            .setEncryptionContext(ENCRYPTION_CONTEXT)
+            .build();
+
+        // Malformed Arn for a non KMS provider shouldn't fail
+        encryptedDataKeys.clear();
+        encryptedDataKeys.add(new KeyBlob("OtherProviderId", "arn:badArn".getBytes(PROVIDER_ENCODING), new byte[]{}));
+        assertFalse(keyring.onDecrypt(decryptionMaterials, encryptedDataKeys).hasCleartextDataKey());
+    }
+
+    @Test
+    void testNullDao() {
+        assertThrows(
+            NullPointerException.class,
+            () -> new AwsKmsSymmetricKeyring(null, KEY_NAME),
+            "dataKeyEncryptionDao is required");
+    }
+
+    @Test
+    void testNullKeyName() {
+        assertThrows(
+            NullPointerException.class,
+            () -> new AwsKmsSymmetricKeyring(dataKeyEncryptionDao, null),
+            "keyName is required");
+    }
+
+    @Test
+    void testEncryptDecryptExistingDataKey() {
+        EncryptionMaterials encryptionMaterials = EncryptionMaterials.newBuilder()
+            .setAlgorithm(ALGORITHM_SUITE)
+            .setCleartextDataKey(PLAINTEXT_DATA_KEY)
+            .setEncryptionContext(ENCRYPTION_CONTEXT)
```

Comment:
I think I might be misunderstanding this. I believe this makes sense as-is, but happy to clarify further.

Reply:
nvm, this is fine as is. I was assuming that we might want to start out with materials that already contain an EDK, because that's how our provided keyrings work, but there is nothing preventing someone from creating a keyring which *only* generates, in which case this is a valid set of materials that we should test.